src/main/org/pantsbuild/jarjar/misplaced/FatalMisplacedClassProcessor.java

```diff
@@ -0,0 +1,33 @@
+package org.pantsbuild.jarjar.misplaced;
+
+import org.pantsbuild.jarjar.util.EntryStruct;
+import org.pantsbuild.jarjar.util.ProcessingContext;
+
+/**
+ * Fails-fast with an exception upon finding a misplaced class.
+ */
+public class FatalMisplacedClassProcessor extends MisplacedClassProcessor {
+
+  public FatalMisplacedClassProcessor(ProcessingContext context) {
+    super(context);
+  }
+
+  @Override public void handleMisplacedClass(EntryStruct classStruct, String className) {
+    throw this.new FatalMisplacedClassException(formatMessage(classStruct, className));
+  }
+
+  @Override public boolean shouldTransform() {
+    return false;
+  }
+
+  @Override public boolean shouldKeep() {
+    return false;
+  }
+
+  class FatalMisplacedClassException extends RuntimeException {
```

Comment:
Also, its not common to raise a RuntimeException. This means that callers can't tell when it is thrown. I would make this a subclass of Exception so you can use a checked exception.